### PR TITLE
[SPARK-46095][DOCS] Document `REST API` for Spark Standalone Cluster

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to document `REST API` for Spark Standalone Cluster.

### Why are the changes needed?

To help the users to understand Apache Spark features.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review. `REST API` Section is added newly.

**AFTER**

Screenshot 2023-11-24 at 4 13 53 PM

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this documentation PR, @HyukjinKwon ?

File: docs/spark-standalone.md [hunk 1]
```diff
@@ -540,6 +542,84 @@ failing repeatedly, you may do so through:
...
+    <td><code>clear</code></td>
+    <td>Clear the completed drivers and applications.</td>
+    <td>POST</td>
+    <td>4.0.0</td>
+  </tr>
+</table>
+
+The following is a <code>curl</code> CLI command example with the `pi.py` and REST API.
+
+{% highlight bash %}
```
Comment (HyukjinKwon) [inline docs/spark-standalone.md:590]: no bigdeal but we can actually just use:
`````
```bash

...

```
`````

File: docs/spark-standalone.md [hunk 2]
```diff
@@ -540,6 +542,84 @@ failing repeatedly, you may do so through:
...
+  "mainClass": "org.apache.spark.deploy.SparkSubmit",
+  "environmentVariables": { },
+  "action": "CreateSubmissionRequest",
+  "appArgs": [ "/opt/spark/examples/src/main/python/pi.py", "10" ]
+}'
+{% endhighlight %}
+
+The following is the response from the REST API for the above <code>create</code> request.
+
+{% highlight bash %}
```
Comment (HyukjinKwon) [inline docs/spark-standalone.md:612]: and 

``````
```

...

```
``````

File: docs/spark-standalone.md [hunk 3]
```diff
@@ -518,6 +518,8 @@ Spark applications supports the following configuration properties specific to s
 
 # Launching Spark Applications
 
+## Spark protocol
```
Comment (HyukjinKwon) [inline docs/spark-standalone.md:521]:
```suggestion
## Spark Protocol
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you. I addressed the comments and verified the result manually.
Merged to master.

**Pull request closed without merging**